### PR TITLE
Expand button

### PR DESCRIPTION
**Pull request by JackRyan1989**
Rationale:

1.  For pages with a large number of accordions (either actual accordion components or details elements) we would like a way for users to easily open and close all accordions.

Changes:

1.  Webpack config has been updated to handle multiple entry points
2.  The details plugin in `/_plugins/details.rb` now sets a data attribute initially. The data attribute is observed by the JS.
3.  New Javascript in `/_assets/js/utils/expand-accordions.js`
4.  New `expand-accordions.html` file in the `_includes` folder. 

Test:

1.  Make sure the expand button doesn't expand any accordions outside of the main content area.
2.  Confirm behavior on mobile (iOS, Android)
3.  Confirm behavior across browsers (Chrome, Firefox, Edge, Safari)
4.  This component needs a11y review from Terri, so do not merge until it has been thoroughly reviewed and tested.

### CODE REVIEW

**Comment by JackRyan1989** [comment]
Here's the full link to test: https://federalist-1716bf9c-dd79-4d96-8285-6e56dc391b84.app.cloud.gov/preview/usdoj-crt/beta-ada/expand-toggle/service-animals-qa/

**Comment by chinelo-ikejimba** [comment]
@JackRyan1989 @c2nelson 

I tried out the full link to test and it works! Well done with adding in the expand/close feature team.

**Comment by JackRyan1989** [comment]
@chinelo-ikejimba Yay! Thanks for reviewing it. Is the current styling ok or do changes need to be made? I think we should talk about placement, maybe?

**Comment by chinelo-ikejimba** [comment]
@JackRyan1989 yeah its placement would be like below

---

Screen Shot 2022-04-06 at 11 24 43 AM

**Comment by JackRyan1989** [comment]
@chinelo-ikejimba Here are some mocks, notice how the text wraps when the title gets longer. This layout is how it looks on mobile as well. 
One other thing, the button is not sticky. So if folks scroll away, the button disappears. I think we may want to make it stay in view, but what are your thoughts?

Screen Shot 2022-04-06 at 4 23 38 PM
Screen Shot 2022-04-06 at 4 24 17 PM
Screen Shot 2022-04-06 at 4 24 52 PM

**Comment by chinelo-ikejimba** [comment]
@JackRyan1989 

Great mockups and questions.

**Regarding the title**
Let's manage the text wrapping on a case-by-case basis. Ideally section titles for FAQ pages would be brief but if they extend to 2 lines or more, then we can look into breaking the title into different typography headings.

**Regarding button placement**
Yes the button's location has tradeoffs when all of the boxes are expanded but its current placement is our best option and a common practice I've seen on other sites like [gov.uk/coronavirus](https://www.gov.uk/coronavirus)'s "Show all sections" button.

**Comment by JackRyan1989** [comment]
@chinelo-ikejimba I have right aligned the button. Please check. If the design looks good then we can move on to accessibility testing with Terri.

**Comment by chinelo-ikejimba** [comment]
Well done with the alignment @JackRyan1989. It looks good to go !

**Comment by JackRyan1989** [comment]
Add aria-controls and aria-expanded to the button as per Terri's request

File: _assets/js/utils/expand-accordions.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+// Grab all of the accordion buttons except for the table of contents button and convert the Nodelist into an array:
+const getAccordionButtons = () => {
+  const buttons = Array.from(document.querySelectorAll('.expand button.usa-accordion__button.pa11y-skip'));
+  return buttons;
+};
+
+// Grab all of the <details> elements and convert the Nodelist into an array:
+const getDetails = () => {
+  let details = Array.from(document.querySelectorAll('details.expand'));
```
Comment (billyhunt) [inline _assets/js/utils/expand-accordions.js:27]: Can be simplified to

```
const getDetails = () => Array.from(document.querySelectorAll('details.expand'));
```

Orrrr change let to const if you want to keep it as is.

File: _assets/js/utils/expand-accordions.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    button.innerText = 'Close all sections';
+    return true;
+  }
+};
+
+// When we click the expand or close button, loop over the accordions and their buttons and either hide them or show them depending on the state of the button
+const expandAccordions = () => {
+  openAccordionsButton.addEventListener('click', function (e) {
+    e.preventDefault();
+    let expanded = toggleButtonText(openAccordionsButton);
```
[the remark this answers is not in the document]
Comment (billyhunt) [reply _assets/js/utils/expand-accordions.js:60]: Don't forget to change your immutable lets to consts.

**Pull request merged**